CHANGELOG.md

```diff
@@ -1,5 +1,310 @@
+      ([#486](https://github.com/cossacklabs/themis/pull/486))
+
+    - EC key generator could produce an invalid public key without returning an error.
+      Thanks to [**@vixentael**](https://github.com/vixentael) for finding this issue.
+      ([#500](https://github.com/cossacklabs/themis/pull/500))
+
+  - **Secure Cell**
+
+    - Improved handling of edge cases with large data buffers.
+      ([#496](https://github.com/cossacklabs/themis/pull/496),
+       [#497](https://github.com/cossacklabs/themis/pull/497))
+
+    - Minor performance improvements in encryption/decryption code path.
+      ([#496](https://github.com/cossacklabs/themis/pull/496))
+
+  - **Library ABI**
+
+    - Themis shared libraries now have a formal ABI version.
+      This ensures that no compatibility issues arise if we ever need to introduce breaking changes in ABI.
+      ([#454](https://github.com/cossacklabs/themis/pull/454))
+
+    - Remove private symbols from public export lists.
+      Themis has been accidentally exporting various private utility functions.
+      Private functions not intended for public use are now kept hidden.
+      ([#458](https://github.com/cossacklabs/themis/pull/458),
+       [#472](https://github.com/cossacklabs/themis/pull/472))
+
+  - <a id="0.12.0-packaging-updates">**Installation & packaging**</a>
+
+    - Themis is now installed to `/usr/local` by default when building from source.
+      ([#448](https://github.com/cossacklabs/themis/pull/448))
+
+      This _may_ be a **breaking change** if you system has non-standard precedence rules.
+      If you install Themis from source code directly, please upgrade cleanly like this:
+
+      ```bash
+      make uninstall PREFIX=/usr
+      make install   PREFIX=/usr/local
+      ```
+
+      Please consider using [binary repositories](https://docs.cossacklabs.com/pages/documentation-themis/#installing-themis-from-repositories) to install Themis.
+      If your system or package manager is not supported yet, please let us know via dev@cossacklabs.com.
+
+    - Themis packages now support multiarch installations.
+      ([#512](https://github.com/cossacklabs/themis/pull/512))
+
+      Multiarch enables parallel installation of 32-bit and 64-bit versions of the library.
+      This is particularly important on CentOS where some tools like _pkg-config_ failed to locate Themis due to non-standard installation path.
+
+    - Makefile now supports a number of [standard GNU variables](https://www.gnu.org/prep/standards/html_node/Directory-Variables.html) like `libdir`.
+      ([#453](https://github.com/cossacklabs/themis/pull/453),
+       [#455](https://github.com/cossacklabs/themis/pull/455))
+
+    - Accurate dependencies for development packages make sure you don't have to install anything manually after installing Themis from package repositories.
+      ([#446](https://github.com/cossacklabs/themis/pull/446))
+
+    - NSIS installer is now available for Windows.
```

Comment:
add smth about this in `TLDR` section? it might be a large change for some of our users